test/metabase/test/redefs.clj

```diff
@@ -29,6 +62,13 @@
         (next-method model attributes f)))
     (next-method model attributes f)))
 
+(defmacro with-temp!-persisted
```

Comment:
Have you thought about breaking this up into orthogonal parts? i.e. a `with-cleanup-disabled` macro that just updates the binding. Two advantages I see:

1. More obvious that the behavior will cascade into nested `with-temp!` calls.
2. Avoid the awkward naming of "temporary but not really". A good enough name to go into toucan2 IMHO.

I guess one thing you'd lose is that "visibility in search" aspect, but I think that can be readily compensated for by mentioning this utility in the `with-temp!` docstring.

Reply:
>Another downside is that wrapping the form will trigger indentation and that's annoying, and could lead to whitespace changes to rollback after unwrapping.

Hopefully you just hit `paredit-raise` on the `with-temp!` form and it removes the `with-cleanup-disabled` and fixes the indentation all at once.

Or, if you're feeling saucy,

```
(with-cleanup-diabled (with-temp! ...
   ...))
```

Reply:
Yeah I'd probably do that one liner.

The issue I'm talking about is that when I surround a form in my editor, it triggers reformatting for the inner forms. This would then wipe out any manual alignment, e.g. that done for the multiple columns in a multi-line `with-temp!`...

Reply:
I wrote a test which required 7 nested macros and I'm thinking if we should invent some kind of `application macro` which can then wrap body in list of somethings. Like `with-open`, maybe called `with-wrap`? Something like Python's `with`, which will call `__enter__` and `__exit__` on a provided object.